core/src/avm2/globals/flash/display/loader_info.rs

```diff
@@ -279,18 +279,23 @@ pub fn get_child_allows_parent<'gc>(
             // TODO: respect allowDomain() and polices.
             avm2_stub_getter!(activation, "flash.display.LoaderInfo", "childAllowsParent");
 
-            let loader = loader_info.loader().expect("Loader should be Some");
-            let loader = loader.as_display_object().expect("Loader is a DO");
-            let parent_movie = loader.movie();
-
-            if let Ok(child_url) = Url::parse(root.url()) {
-                if let Ok(parent_url) = Url::parse(parent_movie.url()) {
-                    if child_url.host() == parent_url.host() {
-                        return Ok(true.into());
+            if let Some(loader) = loader_info.loader() {
+                let loader = loader.as_display_object().expect("Loader is a DO");
+                let parent_movie = loader.movie();
+
+                if let Ok(child_url) = Url::parse(root.url()) {
+                    if let Ok(parent_url) = Url::parse(parent_movie.url()) {
+                        if child_url.host() == parent_url.host() {
+                            return Ok(true.into());
+                        }
                     }
                 }
+                Ok(false.into())
+            } else {
+                // Only the root movie is LoaderStream::Swf but missing a loader.
+                // In that case, return true.
```

Comment:
As mentioned on Discord, I think we should explicitly check if the SWF is the root SWF, and then unwrap the loader and handle the case when a non-root SWF doesn't have a loader.